common/src/main/java/org/opensearch/ml/common/model/MLGuard.java

```diff
@@ -0,0 +1,133 @@
+package org.opensearch.ml.common.model;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+@Log4j2
+@Getter
+public class MLGuard {
+    private Map<String, List<String>> stopWordsIndicesInput = new HashMap<>();
+    private Map<String, List<String>> stopWordsIndicesOutput = new HashMap<>();
+    private List<String> inputRegex;
+    private List<String> outputRegex;
+    private NamedXContentRegistry xContentRegistry;
+    private Client client;
+
+    public MLGuard(Guardrails guardrails, NamedXContentRegistry xContentRegistry, Client client) {
+        this.xContentRegistry = xContentRegistry;
+        this.client = client;
+        if (guardrails == null) {
+            return;
+        }
+        Guardrail inputGuardrail = guardrails.getInputGuardrail();
+        Guardrail outputGuardrail = guardrails.getOutputGuardrail();
+        if (inputGuardrail != null) {
+            fillStopWordsToMap(inputGuardrail, stopWordsIndicesInput);
+            inputRegex = inputGuardrail.getRegex() == null ? new ArrayList<>() : Arrays.asList(inputGuardrail.getRegex());
+        }
+        if (outputGuardrail != null) {
+            fillStopWordsToMap(outputGuardrail, stopWordsIndicesOutput);
+            outputRegex = outputGuardrail.getRegex() == null ? new ArrayList<>() : Arrays.asList(outputGuardrail.getRegex());
+        }
+    }
+
+    private void fillStopWordsToMap(@NonNull Guardrail guardrail, Map<String, List<String>> map) {
+        List<StopWords> stopWords = guardrail.getStopWords();
+        if (stopWords == null || stopWords.isEmpty()) {
+            return;
+        }
+        for (StopWords e : stopWords) {
+            map.put(e.getIndex(), Arrays.asList(e.getSourceFields()));
+        }
+    }
+
+    public Boolean validate(String input, int type) {
+        switch (type) {
```

Comment:
The role of the type parameter isn't immediately clear if you just look at the function definition and its usage. It could be improved by using Java’s Enum to make it self-explanatory.

Reply:
I just use it to differentiate input and output now.

Reply:
Changed to enum in another pr.